internal/server/middleware/grpc/middleware.go

```diff
@@ -320,6 +314,21 @@ func AuditUnaryInterceptor(logger *zap.Logger) grpc.UnaryServerInterceptor {
 			event = audit.NewEvent(audit.Metadata{Type: audit.Namespace, Action: audit.Update, Actor: actor}, r)
 		case *flipt.DeleteNamespaceRequest:
 			event = audit.NewEvent(audit.Metadata{Type: audit.Namespace, Action: audit.Delete, Actor: actor}, r)
+		case *authrpc.CreateTokenRequest:
+			event = audit.NewEvent(audit.Metadata{Type: audit.Token, Action: audit.Create, Actor: actor}, r)
+		case *authrpc.DeleteAuthenticationRequest:
+			a, err := authGetter.GetAuthenticationByID(ctx, r.Id)
+			if err != nil {
+				return nil, err
+			}
+			if a.Method == authrpc.Method_METHOD_TOKEN {
+				event = audit.NewEvent(audit.Metadata{Type: audit.Token, Action: audit.Delete, Actor: actor}, a.Metadata)
+			}
```

Comment:
I would be tempted to do what you suggested the other day and call the type `Authentication` and stash the method in the metadata.
Then nest the authentication metadata in another child called metadata.

```suggestion
			metadata := map[string]string{
				"method":   a.Method,
				"metadata": a.Metadata,
			}
			event = audit.NewEvent(audit.Metadata{Type: audit.Authentication, Action: audit.Delete, Actor: actor}, metadata)
```